gitea/repo.go

```diff
@@ -47,6 +47,25 @@ type Repository struct {
 	Permissions *Permission `json:"permissions,omitempty"`
 }
 
+// Collaborator represents a collaborator
+// swagger:model
+type Collaborator struct {
```

Comment:
Why not use
```Go
// Collaborator represents a user with collaboration details.
type Collaborator struct {
	*User
	Collaboration *Collaboration
}
```

Reply:
ping @zeripath

Reply:
oh I'd sorta given up on this PR. I don't actually need it with the Sudo approach.

If you think the PR is still worth doing I'll do this.

Reply:
I think it might be useful to see which permissions collaborators have, but I have no strong opinion, and I was going through PRs marked as stale so I'm ok with this (and linked PR) being closed.

Reply:
I'll revise, rebase and update